### ISSUE TITLE
isapprox generating macro?

### ISSUE THREAD

**Issue by matthijscox** [schlichtanders/StructEquality.jl#3]
Hi Stephan,

Next to the `==` and `isequal` for structs, I'm always writing `isapprox` functions for my structs that contain floating point fields. One day I am going to write a macro for that. Would this package be a good place for it?

It's a bit more complex maybe, due to combinations of types that need `isapprox` and those that need `isequal`.

Here's what it always roughly looks like:
```julia
struct MyStruct
  a::Float64
  b::Vector{Float64}
  c::Int
end

function Base.isapprox(m1::MyStruct, m2::MyStruct; kwargs...)
    return isapprox(m1.a, m2.a; kwargs...) &&
           isapprox(m1.b, m2.b; kwargs...) &&
           isequal(m1.c, m2.c)
end
```

**Comment by schlichtanders** [comment]
Hi @matthijscox thank you for the idea and the connection to StructEquality.
It took some weeks, but just now I have it implement in the `develop` branch, tests are created and running, but I still need to write documentation for it be pushed to the next version.

I renamed the macros and the one for isapprox is called `@struct_isapprox`.

It would be great if you could give it a try and test whether things are working as intended
```
] dev https://github.com/schlichtanders/StructEquality.jl#develop
```

**Comment by matthijscox** [comment]
Seems to work so far:

```julia
using StructEquality
using Test

@struct_isapprox Base.@kwdef struct Foo{T}
    name::String = "myfoo"
    value::T = 5.0
end

@struct_isapprox Base.@kwdef struct Bar{T}
    name::String = "mybar"
    foo::Foo = Foo()
    value::T = 1.0
end

b = Bar()
b2 = Bar()
@test b ≈ b2

b = Bar()
b2 = Bar(value = 1.1)
@test isapprox(b, b2, atol=0.11)
@test isapprox(b, b2, rtol=0.11)

b = Bar()
b2 = Bar(value = 2.0)
@test !(b ≈ b2)

b = Bar()
b2 = Bar(foo = Foo(value = -1.0))
@test !(b ≈ b2)

```

**Comment by schlichtanders** [comment]
nice, thank you for the test case

I am going to include it into the code if this is okay with you

**Comment by matthijscox** [comment]
> I am going to include it into the code if this is okay with you

Sure, go ahead

I also quickly tested that I can still add docstrings
```julia
"""
Can I still add docstrings?
"""
@struct_isapprox Base.@kwdef struct Bar{T}
    name::String = "mybar"
    foo::Foo = Foo()
    value::T = 1.0
end
Base.doc(Bar)
```

On the REPL this indeed shows me the doc string. I don't know a way to easily test this.

**Comment by matthijscox** [comment]
I see you released this functionality in v2.0.0. Awesome!

**Comment by schlichtanders** [comment]
thank you :slightly_smiling_face:
